csharp/private/actions/assembly.bzl

```diff
@@ -102,6 +102,17 @@ def AssemblyAction(
 
     args.add("/debug:portable")
 
+    # The full path of source is embedded into PDBs when compiled
+    # To resolve this, we use pathmap to substitute in new paths
+    # that ensure that these are deterministic
+    #
+    # The dotnet wrapper (dotnetw) provides pathing variables
+    # that can be substituted into any arguments passed in
+    # These are of the form: __BAZEL_{name}_
+
+    # pathmap
+    args.add("/pathmap:%s=%s" % ("__BAZEL_SANDBOX__", "__BAZEL_WORKSPACE__"))
```

Comment:
Why pass this arg at all (and why the string formatting?)

Reply:
I wrote this line with the idea in mind that we may need multiple `/pathmap` statements, which would be something like: `args.add_all(pathmaps, map_each = _format_pathmap_arg)`. Although I have checked with our examples, I'll likely take a look at a couple more cases to see if we need anymore statements.

The processor variables (`__BAZEL_<>_`) are so the wrapper can substitute context variables (e.g. filesystem paths) into the command line parameters.

The strings are constant, and resolve to the following: `/pathmap:__BAZEL_SANDBOX__=__BAZEL_WORKSPACE__`.

Reply:
Ah, I can't think of a scenario where we'd pass multiple `/pathmap`. It's something that `dotnet.exe` supports but I don't think we'd have a use-case (for us we just need determinism, and one pathmap should be sufficient for that).